lang/src/org/partiql/lang/syntax/SqlParser.kt

```diff
@@ -1878,15 +1875,134 @@ class SqlParser(private val ion: IonSystem) : Parser {
+                }
+                else {
+                    rem.parseExpression(precedence)
+                }
+            }
+        }
+        rem = child.remaining
+
+        child = rem.parseOptionalAsAlias(child).also {
+            rem = it.remaining
+        }
+
+        child = rem.parseOptionalAtAlias(child).also {
+            rem = it.remaining
+        }
+
+        child = rem.parseOptionalByAlias(child).also {
+            rem = it.remaining
+        }
+
+        var left = child
+
+        var delim = rem.parseJoinDelim()
+        if (parseRemaining) {
+            while (delim?.type?.isJoin == true) {
+                val isCrossJoin = delim.token?.keywordText?.contains("cross") ?: false
+                val hasOnClause = delim.token?.type == KEYWORD && !isCrossJoin
+                var children : List<ParseNode>
+                var joinToken : Token? = delim.token
+
+                rem = rem.tail
+
+                if (hasOnClause) {
+                    // Explicit join
+                    if (rem.head?.type == LEFT_PAREN) {
+                        // Starts with a left paren. Could indicate subquery/literal or indicate higher precedence
+                        val isSubqueryOrLiteral = rem.tail.head?.type == LITERAL || rem.tail.head?.keywordText == "select"
+                        val parenClause = rem.parseFromSource(precedence, true)
+                        rem = parenClause.remaining
+
+                        // check for an ON-clause
+                        if (rem.head?.keywordText != "on") {
+                            rem.err("Expected 'ON'", PARSE_MALFORMED_JOIN)
+                        }
+
+                        val onClause = rem.tail.parseExpression(precedence)
+
+                        rem = onClause.remaining
+                        if (!isSubqueryOrLiteral) {
+                            children = listOf(parenClause, left, onClause)
+                        }
+
+                        else {
+                            children = listOf(left, parenClause, onClause)
+                        }
+                    }
+
+                    else {
+                        // Rest is just the right side of the clause
+                        val rightRef = rem.parseFromSource(precedence, false)
+                        rem = rightRef.remaining
+
+                        // check for an ON-clause
+                        if (rem.head?.keywordText != "on") {
+                            rem.err("Expected 'ON'", PARSE_MALFORMED_JOIN)
+                        }
+
+                        val onClause = rem.tail.parseExpression(precedence)
+
+                        rem = onClause.remaining
+
+                        children = listOf(left, rightRef, onClause)
+                    }
+                }
+
+                else {
+                    // For implicit joins
+                    val rightRef = rem.parseFromSource(precedence, false)
```

Comment:
use named arguments for when passing boolean literals.

Reply:
Changed.